File: JavaTest/EnumerationTest.java
import MyLibrary.*;
import org.junit.*;

import static org.junit.Assert.*;

public class EnumerationTest extends TestBase {
    @Test
    public void EnumValuesTest() {
        EnumerationTypeExample e = EnumerationTypeExample.Label2;
        assertEquals(2, EnumerationUserExample.ConvertToInteger(e));
        assertEquals(e, EnumerationUserExample.ConvertToEnumeration(2));
    }
}

File: CSharpTest/EnumerationTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CSharpTest
{
    using E = MyLibrary.EnumerationTypeExample;
    using C = MyLibrary.EnumerationUserExample;

    [TestClass]
    public class EnumerationTest
    {
        [TestMethod]
        public void EnumValuesTest()
        {
            var e = E.Label2;
            Assert.AreEqual(2, C.ConvertToInteger(e));
            Assert.AreEqual(e, C.ConvertToEnumeration(2));
        }
    }
}
